### ISSUE TITLE
Can't assign function with non keyword-only params to value expecting function with keyword-only params

### ISSUE THREAD

**Issue by Tomaz-Vieira** [microsoft/pyright#2048]
**Describe the bug**
Assigning a function that has normal parameters (i.e. parameters that are not keyworkd-only) to a value that expects a function with keyword-only arguments produces an error.

**To Reproduce**
```python3
# pyright: strict

from typing import Protocol

class MyCallable(Protocol):
    def __call__(self, *, a: int, b: str) -> int: ...

def my_function(a: int, b: str) -> int:
    return 123

# Expression of type "(a: int, b: str) -> int" cannot be assigned to declared type "MyCallable"
# "(a: int, b: str) -> int" cannot be assigned to type "(*, a: int, b: str) -> int"
f : MyCallable = my_function 
```

**Expected behavior**
No errors when assigning `my_function` of variable of type `MyCallable`.

**VS Code extension or command-line**
pyright extension version 1.1.150
cli tool version 1.1.153

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.155, which I just published. It will also be included in the next release of pylance.
